src/Operation/Unfold.php

```diff
@@ -25,30 +25,29 @@ final class Unfold extends AbstractOperation
     /**
      * @pure
      *
-     * @return Closure(T...): Closure(callable(mixed|T...): (mixed|array<TKey, T>)): Closure(): Generator<int, T>
+     * @return Closure(T...): Closure(callable(T...): list<T>): Closure(): Generator<int, list<T>>
      */
     public function __invoke(): Closure
     {
         return
             /**
              * @param T ...$parameters
              *
-             * @return Closure(callable(mixed|T...): (array<TKey, T>)): Closure(): Generator<int, T>
+             * @return Closure(callable(T...): list<T>): Closure(): Generator<int, list<T>>
              */
             static fn (...$parameters): Closure =>
                 /**
-                 * @param callable(mixed|T...): (mixed|array<TKey, T>) $callback
+                 * @param callable(T...): list<T> $callback
                  *
-                 * @return Closure(): Generator<int, T>
+                 * @return Closure(): Generator<int, list<T>>
                  */
                 static fn (callable $callback): Closure =>
                     /**
-                     * @return Generator<int, T>
+                     * @return Generator<int, list<T>>
                      */
                     static function () use ($parameters, $callback): Generator {
                         while (true) {
-                            /** @var T $parameters */
-                            $parameters = $callback(...array_values((array) $parameters));
+                            $parameters = $callback(...$parameters);
```

Comment:
I think we should not create a local variable and `yield` directly the result of the callback.

Reply:
that won't work the same way because the Generator uses the variable in the next yield iteration here:
`$callback(...$parameters)` 

If there's no variable then it will continue to use the _initial_ parameters

Reply:
#TripleFacepalm ... damn I'm too tired!

Reply:
> #TripleFacepalm ... damn I'm too tired!

Don't worry about it 😄. I tried it as well haha, had the same idea initially

Reply:
😌😓